axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/ebics/service/EbicsCertificateService.java

```diff
@@ -127,78 +139,52 @@ private byte[] getSSLCertificate(EbicsBank bank) throws AxelorException {
-                  throws CertificateException {
-                accepted = arg0;
-              }
-
-              @Override
-              public X509Certificate[] getAcceptedIssuers() {
-                return accepted;
-              }
-            }
-          },
-          null);
-
-      HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
-
-      HttpsURLConnection.setDefaultHostnameVerifier(
-          new HostnameVerifier() {
+      EbicsUtils.setProxy(builder);
 
+      builder.addInterceptorLast(
+          new HttpResponseInterceptor() {
             @Override
-            public boolean verify(String arg0, SSLSession arg1) {
-              return true;
+            public void process(HttpResponse response, HttpContext context)
+                throws HttpException, IOException {
+              ManagedHttpClientConnection routedConnection =
+                  (ManagedHttpClientConnection)
+                      context.getAttribute(HttpCoreContext.HTTP_CONNECTION);
+              SSLSession sslSession = routedConnection.getSSLSession();
+              if (sslSession != null) {
+                Certificate[] certificates = sslSession.getPeerCertificates();
+                context.setAttribute(CONTEXT_CERTI_KEY, certificates);
+              }
             }
           });
 
-      connection.setSSLSocketFactory(sslCtx.getSocketFactory());
-      log.debug("SSL connection response code: {}", connection.getResponseCode());
-      log.debug("SSL connection response message: {}", connection.getResponseMessage());
+      CloseableHttpClient httpclient = builder.build();
+      HttpContext context = new BasicHttpContext();
+      HttpPost httpPost = new HttpPost(bankUrl.getPath());
+
+      CloseableHttpResponse response = httpclient.execute(targethost, httpPost, context);
 
-      if (connection.getResponseCode() == 200) {
-        Certificate[] certificates = connection.getServerCertificates();
-        for (int i = 0; i < certificates.length; i++) {
-          Certificate certificate = certificates[i];
+      if (response.getStatusLine() != null && response.getStatusLine().getStatusCode() == 200) {
+        Certificate[] peerCertificates = (Certificate[]) context.getAttribute(CONTEXT_CERTI_KEY);
+        for (int i = 0; i < peerCertificates.length; i++) {
+          Certificate certificate = peerCertificates[i];
           if (certificate instanceof X509Certificate) {
             X509Certificate cert = (X509Certificate) certificate;
             createCertificate(cert, bank, EbicsCertificateRepository.TYPE_SSL);
             return certificate.getEncoded();
           }
         }
       }
-      connection.disconnect();
 
     } catch (Exception e) {
-      e.printStackTrace();
+      TraceBackService.trace(e);
     }
 
     return null;
-
-    //		throw new AxelorException(I18n.get("Error in getting ssl certificate"),
-    // TraceBackRepository.CATEGORY_CONFIGURATION_ERROR);
-
```

Comment:
Please remove this commented out code

Reply:
I have already removed this code 
Please let me know if need to change that

Reply:
you're right, I read the changes wrong, my bad.